### ISSUE TITLE
how to patch oneplus 6 or 6t HID kernel??

### ISSUE THREAD

**Issue by johanlike** [offensive-security/kali-nethunter#1245]
my kernel is 4.9X how to patch HID????

**Comment by simonpunk** [comment]
@johanlike
Try this, https://drive.google.com/file/d/1z-3p1DEkKykqb_wYzYqnXPdmh9ArRTOw/view?usp=drivesdk
but I dont have a device to test.  If you manage to build the kernel with my patch, please do let me know, thanks.

**Comment by johanlike** [comment]
@simonpunk Hello simonpunk, thank you very much for replying to my question. I have been following you on both the XDA forum and github. If the patch is used like this (patch -p1 OP6.hid.patch), unfortunately, this patch cannot be successfully injected. Now I only have the OP6T phone, but the OP6 and OP6T use the same kernel, but the Qualcomm 845 processor. There is no /drivers/usb/gadget/android.c file in the project that builds the kernel. Here is the link to the project I used to build the kernel:
https://github.com/OnePlusOSS/android_kernel_oneplus_sdm845
https://gitlab.com/HolyAngel/op6/tree/oos/drivers/usb/gadget
The following is a screenshot of the injection (it stuck for a very long time at the time of injection):


![1](https://user-images.githubusercontent.com/17888022/52216341-f7f75180-28d0-11e9-8f78-eef036191623.png)
Thank you again for helping me, thank you very much.

**Comment by johanlike** [comment]
@simonpunk I have now fixed the support for the external OTG wireless card and hackrf one, rtl-sdr, and it works fine. It would be great if you were fixing the HID feature. Please contact me if there is anything that can help you. thank you for your help.

**Comment by simonpunk** [comment]
@johanlike shouldn't it be "patch -p1 < OP6.hid.patch" ?? I'd like to know how you managed to build a working kerne, please do let me  know how, thanks.

**Comment by johanlike** [comment]
Oh no, I forgot this symbol, thank you for reminding me that I am passing the Chinese New Year is equivalent to American Christmas. I wish you all the best here, a rich night.Wishing you health and happiness in the year to come @simonpunk

**Comment by johanlike** [comment]
补丁现在可以正常的注入了，晚些时候我会尝试构建的新的内核，到时我会告诉你这个内核是否可用，如果可用我会将这个内核分享给你。再次感谢你的帮助。非常感谢。 @simonpunk

**Comment by johanlike** [comment]
@simonpunk The patch can now be injected normally. Later I will try to build a new kernel. I will tell you if the kernel is available. If it is available, I will share it with you. Thank you again for your help. thank you very much.

**Comment by simonpunk** [comment]
@johanlike Your're welcome, and Happy Chinese New Year too. Just ping me later. :)

**Comment by kimocoder** [comment]
kimocoder is watching.. I also have the flar2 kernel ready for testing soon.
OnePlus 6 support would be awesome!

**Comment by WEIWULUNLUN** [comment]
@kimocoder @simonpunk   @johanlike新年快乐

**Comment by johanlike** [comment]
@simonpunk Hello simonpunk If I open the HID attack with kali's official nethunter app, I will have this file in /dev/hidg0. Even if I give 666 permission, it will not work properly, but if you open the HID attack with the nethunter compiled on github, there is no Hidg* this file, what is the cause of this?Thank you for your help and wish you a happy life.

**Comment by johanlike** [comment]
@simonpunk  Hello simonpunk  If I open the HID attack with kali's official nethunter app, /dev/hidg0 has this file in this location, even if I give 666 permission, it will not work properly, but if you open the HID attack with the nethunter compiled on github, there is no Hidg* this file, what is the reason for this? Thank you again for helping me, I wish you good luck every day.

**Comment by simonpunk** [comment]
@johanlike sorry for late reply. And sorry I forgot to mention that you have to define a service for hid function first. If you dont have my init.nethunter.rc installed in android root folder, it wont activate the function. You can downlod my OP5 kernel zip in my google drive and extract that rc file, you will see how it works. 

But there is still a way to achieve this. If you are using my nethunter-app, in the USB Army fragment, you can run the switch_on_HID.sh script in the script runner section, it should be able to bring up the HID function as well as the /dev/hidg0-1, the script may be depreciated though as I havent tested it for a long time.

**Comment by johanlike** [comment]
@simonpunk Hello, you can reply to me, I am already very happy, and you have provided me with help, I don't mind waiting for the length of time. Because I am in China, you are in a distant country, because of the geographical location, we may not have the same time and life. Here is a screenshot of my running switch_on_HID.sh:
![666](https://user-images.githubusercontent.com/17888022/52341955-04a1b400-2a4f-11e9-810a-1916c032fa58.jpg)
![888](https://user-images.githubusercontent.com/17888022/52341970-13886680-2a4f-11e9-9a85-5fe4ce78047f.png)
Even though I installed the init.nethunter.rc file to the Android root directory, it didn't work. Here is my screenshot:
![999](https://user-images.githubusercontent.com/17888022/52342073-56e2d500-2a4f-11e9-817d-2517b6d466e4.png)
![777](https://user-images.githubusercontent.com/17888022/52342080-5f3b1000-2a4f-11e9-8e9c-e00eda67145f.jpg)
I am very grateful that you can help, I am happy to share my zip copy and compile the configuration file. Here is my file compression package:
https://drive.google.com/file/d/1AWtVr-DZ_hBgzFjVJVPkfYrlBcML5vKc/view
Here is the link to the project I used to build the kernel:
https://gitlab.com/HolyAngel/op6
Please contact me if you have any questions, I will try my best to help. Thank you again for your help. Thank you。

**Comment by simonpunk** [comment]
@johanlike Do you have telegram? I think it's better to talk in telegram as there are few things I want to explain to you and see if you did it in a wrong way. You can find my telegram id from my signature of my XDA thread. Thanks.

**Comment by johanlike** [comment]
@simonpunk hello simonpunk，I think so too. I just finished watching the movie. Now we are late at night. Please wait for me to add your telegram immediately.

**Comment by netnetnetnetnet** [comment]
@johanlike 啥是hid ，nethunter支持6t吗？

**Comment by johanlike** [comment]
@simonpunk Hello simonpunk 。I have seen your latest kernel, and I will test it right away. It's a pity that I added your telegram and can't send you a message because the national telegram does not allow people in my country to send messages overseas. I will try to appeal to the telegraph official. thank you for your help.

**Comment by johanlike** [comment]
@netnetnetnetnet 简单来讲就是硬件攻击，新年快乐。

**Comment by netnetnetnetnet** [comment]
@johanlike 谢谢你回复。你给一加6刷nethunter成功了吗？能不能给我你的微信号，我想问一下。

**Comment by johanlike** [comment]
@netnetnetnetnet https://t.me/nethunter666 电报链接。

**Comment by WEIWULUNLUN** [comment]
@johanlike完整的内核只有14M嘛？我想在我的op6上测试你的内核。

**Comment by johanlike** [comment]
@WEIWULUNLUN 上面有simonpunk的修改后的版本，那是我的V2版本他修改后算是V3了，今天凌晨我和他一直在探讨怎么修补。在他的帮助下能成功调出HID并在我的kali linux 电脑leafpad上面输出数据，可是应用程序有点BUG还需要完善。人很好，而且非常耐心的帮助我修补和测试，真是十分感激他。再次感谢 @simonpunk Thank you again for your help. Thank you。

**Comment by acai66** [comment]
Have you compiled the kernel to add USB WiFi support? I'm using oneplus 6 and looking forward to your great jobs.

**Comment by johanlike** [comment]
@acai66 Yes, I have completed WIFI and SDR support and are now completing the final HID fix. If you can't wait, you can use the kernel above.

**Comment by kimocoder** [comment]
Any progress here? I'll be happy to try the kernel 🙂

**Comment by johanlike** [comment]
Yes, I have fixed the HID function and the nethunter app. It is expected to be released to XDA and github two days later. The demo video can be viewed at https://t.me/nethunter666 @simonpunk @kimocoder @WEIWULUNLUN @netnetnetnetnet @acai66 Thank you again for your help. Thank you@simonpunk

**Comment by johanlike** [comment]
@kimocoder @WEIWULUNLUN @netnetnetnetnet @acai66 https://github.com/johanlike/Oneplus6-or-6T-Nethunter-Kernel XDA:https://forum.xda-developers.com/oneplus-6t/themes/oneplus6-6t-nethunter-kernel-t3900105

**Comment by flhang** [comment]
@johanlike 这个可以完美使用了么， 可以的话， 我去买个oneplus6t 。

**Comment by johanlike** [comment]
@flhang  已经可以完美使用具体可以看我的主页，我会不定期更新.

**Comment by flhang** [comment]
@johanlike 我晕， 我刚买了一个， 你就把github给删了？

**Comment by johanlike** [comment]
@flhang 我也不想删，但是有些人叽叽歪歪的说着说那。我辛辛苦苦编译免费开放给他们用，没得句谢谢感激之类的好话，还要倒打一钯，你说这让我怎么能释怀，项目我已经停止更新，如果你需要下载可以到我提交历史哪里点进去下载，或者到别人分支下载。

**Comment by flhang** [comment]
@johanlike  多好的项目啊。。。，删了可惜了。  哪个分支可以下载， 我记得还有一大堆操作说明，
